packages/zoe/src/instanceRecordStorage.js

```diff
@@ -1,142 +1,103 @@
+ * @returns {(ir: InstanceRecord) => InstanceState}
  */
 export const makeInstanceRecordStorage = baggage => {
   provide(baggage, 'instanceRecord', () => undefined);
 
-  const assertInstantiated = () =>
+  const assertInstantiated = instanceRecord =>
     assert(
-      baggage.get('instanceRecord') !== 'undefined',
+      instanceRecord !== 'undefined',
       'instanceRecord has not been instantiated',
     );
 
-  const addIssuerToInstanceRecord = (keyword, issuerRecord) => {
-    !(keyword in issuerRecord) || Fail`conflicting definition of ${q(keyword)}`;
-
-    assertInstantiated();
-    const instanceRecord = baggage.get('instanceRecord');
-    const nextInstanceRecord = harden({
-      ...instanceRecord,
-      terms: {
-        ...instanceRecord.terms,
-        issuers: {
-          ...instanceRecord.terms.issuers,
-          [keyword]: issuerRecord.issuer,
-        },
-        brands: {
-          ...instanceRecord.terms.brands,
-          [keyword]: issuerRecord.brand,
-        },
-      },
-    });
-    baggage.set('instanceRecord', nextInstanceRecord);
-  };
-
-  /** @type {GetInstanceRecord} */
-  const getInstanceRecord = () => {
-    assertInstantiated();
-    return baggage.get('instanceRecord');
-  };
-  const getTerms = () => {
-    assertInstantiated();
-    return baggage.get('instanceRecord').terms;
-  };
-  const getIssuers = () => {
-    assertInstantiated();
-    return baggage.get('instanceRecord').terms.issuers;
-  };
-  const getBrands = () => {
-    assertInstantiated();
-    return baggage.get('instanceRecord').terms.brands;
-  };
-  /** @type {InstanceRecordManagerGetInstallationForInstance} */
-  const getInstallationForInstance = () => {
-    assertInstantiated();
-    return baggage.get('instanceRecord').installation;
-  };
+  const InstanceRecordInterface = M.interface(
+    'InstanceRecord',
+    // TODO (cth)   actually...
+    {},
+    { sloppy: true },
+  );
 
-  const assertUniqueKeyword = keyword => {
-    assertInstantiated();
-    assertKeywordName(keyword);
-    !ownKeys(baggage.get('instanceRecord').terms.issuers).includes(keyword) ||
-      Fail`keyword ${q(keyword)} must be unique`;
-  };
-
-  const instantiate = startingInstanceRecord => {
-    assert(
-      baggage.get('instanceRecord') === undefined,
-      X`instanceRecord ${baggage.get(
-        'instanceRecord',
-      )} can only be instantiated once`,
-    );
-    baggage.set('instanceRecord', startingInstanceRecord);
-  };
-
-  return harden({
-    addIssuerToInstanceRecord,
-    getInstanceRecord,
-    getTerms,
-    getInstallationForInstance,
-    getIssuers,
-    getBrands,
-    assertUniqueKeyword,
-    instantiate,
-  });
-};
+  const makeInstanceRecord = vivifyFarClass(
+    baggage,
+    'InstanceRecord',
+    InstanceRecordInterface,
+    record => harden({ instanceRecord: record }),
+    {
+      addIssuer(keyword, issuerRecord) {
+        const { state } = this;
+        !(keyword in issuerRecord) ||
```

Comment:
This use of `in` rather than an own property check looks like a bug susceptible to prototype collision, albeit one that is preexisting and appears to fail safe. Some (and perhaps all?) paths to it are protected by `KeywordShape` guards, but I don't see any implementation of the desired claim that each keyword actually starts with an uppercase letter because our pattern library does not cover string contents. Should the check be aligned with `assertUniqueKeyword`?
```suggestion
        !ownKeys(issuerRecord).includes(keyword) ||
```

Reply:
done